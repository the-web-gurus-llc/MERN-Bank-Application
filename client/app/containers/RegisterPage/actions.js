/*
 *
 * RegisterPage actions
 *
 */

import {
  REGISTER_SUCCESS,
  REGISTER_ERROR,
  REGISTER_REQUEST,
  LOGIN_EXPRESS_REQUEST,
  LOGIN_EXPRESS_SUCCESS,
  LOGIN_EXPRESS_ERROR,
} from './constants';

export function registerAction() {
  return {
    type: REGISTER_REQUEST,
  };
}

export function registerSuccessAction(pinCode) {
  return {
    type: REGISTER_SUCCESS,
    pinCode,
  };
}

export function registerErrorAction(error) {
  return {
    type: REGISTER_ERROR,
    error,
  };
}

export function loginExpressAction() {
  return {
    type: LOGIN_EXPRESS_REQUEST,
  };
}

export function loginExpressSuccessAction(user, token) {
  return {
    type: LOGIN_EXPRESS_SUCCESS,
    user,
    token,
  };
}

export function loginExpressErrorAction(error) {
  return {
    type: LOGIN_EXPRESS_ERROR,
    error,
  };
}
