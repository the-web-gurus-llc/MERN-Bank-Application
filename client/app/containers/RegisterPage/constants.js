/*
 *
 * RegisterPage constants
 *
 */

export const REGISTER_REQUEST = 'app/RegisterPage/REGISTER_REQUEST';
export const REGISTER_SUCCESS = 'app/RegisterPage/REGISTER_SUCCESS';
export const REGISTER_ERROR = 'app/RegisterPage/REGISTER_ERROR';

export const LOGIN_EXPRESS_REQUEST = 'app/RegisterPage/LOGIN_EXPRESS_REQUEST';
export const LOGIN_EXPRESS_SUCCESS = 'app/RegisterPage/LOGIN_EXPRESS_SUCCESS';
export const LOGIN_EXPRESS_ERROR = 'app/RegisterPage/LOGIN_EXPRESS_ERROR';
