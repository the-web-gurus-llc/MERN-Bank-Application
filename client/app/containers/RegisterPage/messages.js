/*
 * RegisterPage Messages
 *
 * This contains all the text for the RegisterPage container.
 */

import { defineMessages } from 'react-intl';

export const scope = 'app.containers.RegisterPage';

export default defineMessages({
  description: {
    id: `${scope}.description`,
    defaultMessage: 'Description of RegisterPage',
  },
  registration: {
    id: `${scope}.registration`,
    defaultMessage: 'Registration',
  },
  emailUnique: {
    id: `${scope}.email.unique`,
    defaultMessage: `Email address already exists.`,
  },
  serverError: {
    id: `${scope}.serverError`,
    defaultMessage: `Please try again in a moment.`,
  },
});
