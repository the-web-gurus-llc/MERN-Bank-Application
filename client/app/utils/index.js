export { default as colors } from './colors';
export { default as media } from './media';
export { default as request } from './request';
export { default as typography } from './typography';
export { default as api } from './api';
export { default as routes } from './routes';
export { default as locales } from './locales';
export { default as i18n } from './i18n';
