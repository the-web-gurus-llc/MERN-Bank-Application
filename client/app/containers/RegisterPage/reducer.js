/*
 *
 * RegisterPage reducer
 *
 */

import produce from 'immer';
import { LOCATION_CHANGE } from 'connected-react-router';
import {
  SELECT_CURRENCY,
  NEXT_STEP,
  CHANGE_INPUT,
  PREVIOUS_STEP,
} from 'containers/App/constants';
import { routes } from 'utils';
import { REGISTER_SUCCESS } from './constants';

export const initialState = {
  firstName: '',
  lastName: '',
  currency: null,
  email: '',
  password: '',
  pinCode: '',
  currentStep: 0,
};

/* eslint-disable default-case, no-param-reassign, consistent-return */
const registerPageReducer = produce((draft, action) => {
  if (window.location.pathname === routes.register.path) {
    switch (action.type) {
      case CHANGE_INPUT:
        draft[action.name] = action.value.trim();
        break;
      case SELECT_CURRENCY:
        draft.currency = action.currency;
        break;
      case NEXT_STEP:
        draft.currentStep += 1;
        break;
      case PREVIOUS_STEP:
        draft.currentStep -= 1;
        break;
    }
  }

  switch (action.type) {
    case REGISTER_SUCCESS:
      draft.pinCode = action.pinCode;
      break;
    case LOCATION_CHANGE:
      return initialState;
  }
}, initialState);

export default registerPageReducer;
