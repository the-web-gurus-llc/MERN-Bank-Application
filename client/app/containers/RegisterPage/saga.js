import { call, put, select, takeLatest } from 'redux-saga/effects';
import { api, request, routes } from 'utils';
import React from 'react';
import { FormattedMessage } from 'react-intl';
import { push } from 'connected-react-router';
import messages from './messages';
import { REGISTER_REQUEST, LOGIN_EXPRESS_REQUEST } from './constants';
import {
  registerSuccessAction,
  registerErrorAction,
  loginExpressErrorAction,
  loginExpressSuccessAction,
} from './actions';
import {
  makeSelectFirstName,
  makeSelectEmail,
  makeSelectLastName,
  makeSelectPassword,
  makeSelectCurrency,
  makeSelectPinCode,
} from './selectors';

export function* register() {
  const firstName = yield select(makeSelectFirstName());
  const lastName = yield select(makeSelectLastName());
  const email = yield select(makeSelectEmail());
  const password = yield select(makeSelectPassword());
  const currency = yield select(makeSelectCurrency());

  const requestURL = api.auth.register;
  const requestParameters = {
    method: 'POST',
    headers: { Accept: 'application/json', 'Content-Type': 'application/json' },
    body: JSON.stringify({ firstName, lastName, email, password, currency }),
  };

  try {
    const {
      userAuth: { pinCode },
    } = yield call(request, requestURL, requestParameters);
    yield put(registerSuccessAction(pinCode));
  } catch (error) {
    let message;

    switch (error.statusCode) {
      case 409:
        message = <FormattedMessage {...messages.emailUnique} />;
        break;

      default:
        message = <FormattedMessage {...messages.serverError} />;
        break;
    }

    yield put(registerErrorAction(message));
  }
}

export function* loginExpress() {
  const pinCode = yield select(makeSelectPinCode());
  const password = yield select(makeSelectPassword());

  const requestURL = api.auth.login;
  const requestParameters = {
    method: 'POST',
    headers: {
      Accept: 'application/json',
      'Content-Type': 'application/json',
    },
    body: JSON.stringify({ pinCode, password }),
  };

  try {
    const { user, token } = yield call(request, requestURL, requestParameters);
    yield put(loginExpressSuccessAction(user, token));
    yield put(push(routes.dashboard.path));
  } catch (error) {
    const message = <FormattedMessage {...messages.serverError} />;
    yield put(loginExpressErrorAction(message));
  }
}

export default function* registerPageSaga() {
  yield takeLatest(REGISTER_REQUEST, register);
  yield takeLatest(LOGIN_EXPRESS_REQUEST, loginExpress);
}
