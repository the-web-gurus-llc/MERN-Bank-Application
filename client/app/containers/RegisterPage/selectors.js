import { createSelector } from 'reselect';
import { initialState } from './reducer';

/**
 * Direct selector to the registerPage state domain
 */

const selectRegisterPageDomain = (state) => state.registerPage || initialState;

/**
 * Other specific selectors
 */

const makeSelectFirstName = () =>
  createSelector(selectRegisterPageDomain, (substate) => substate.firstName);

const makeSelectLastName = () =>
  createSelector(selectRegisterPageDomain, (substate) => substate.lastName);

const makeSelectPassword = () =>
  createSelector(selectRegisterPageDomain, (substate) => substate.password);

const makeSelectCurrency = () =>
  createSelector(selectRegisterPageDomain, (substate) => substate.currency);

const makeSelectEmail = () =>
  createSelector(selectRegisterPageDomain, (substate) => substate.email);

const makeSelectPinCode = () =>
  createSelector(selectRegisterPageDomain, (substate) => substate.pinCode);

const makeSelectError = () =>
  createSelector(selectRegisterPageDomain, (substate) => substate.error);

const makeSelectCurrentStep = () =>
  createSelector(selectRegisterPageDomain, (substate) => substate.currentStep);

/**
 * Default selector used by RegisterPage
 */

export {
  selectRegisterPageDomain,
  makeSelectFirstName,
  makeSelectLastName,
  makeSelectPassword,
  makeSelectCurrency,
  makeSelectEmail,
  makeSelectPinCode,
  makeSelectError,
  makeSelectCurrentStep,
};
