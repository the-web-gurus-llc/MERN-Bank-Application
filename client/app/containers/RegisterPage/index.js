/**
 *
 * RegisterPage
 *
 */

import React from 'react';
import { Helmet } from 'react-helmet-async';
import { FormattedMessage } from 'react-intl';
import { useInjectReducer, useInjectSaga } from 'redux-injectors';
import Header from 'components/Header';
import Subheader from 'components/Subheader';
import RegisterForm from 'components/RegisterForm';
import RedirectToggle from 'components/RedirectToggle';
import Footer from 'components/Footer';
import Information from 'components/Information';
import { routes } from 'utils';
import reducer from './reducer';
import saga from './saga';
import messages from './messages';
import ConstantCookie from '../../components/ConstantCookie';

const key = 'registerPage';

export default function RegisterPage() {
  useInjectReducer({ key, reducer });
  useInjectSaga({ key, saga });

  return (
    <>
      <FormattedMessage {...messages.registration}>
        {(title) => <Helmet title={title} />}
      </FormattedMessage>

      <Header />
      <Subheader pageTitle={routes.register.name} />

      <Information />

      <RegisterForm />

      <RedirectToggle />
      <Footer />

      <ConstantCookie />
    </>
  );
}
