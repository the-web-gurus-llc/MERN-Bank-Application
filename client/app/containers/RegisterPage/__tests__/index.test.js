import React from 'react';
import { render } from '@testing-library/react';
import { IntlProvider } from 'react-intl';
import { HelmetProvider } from 'react-helmet-async';
import { createMemoryHistory } from 'history';
import { Provider } from 'react-redux';
import { ConnectedRouter } from 'connected-react-router';
import configureStore from 'utils/configureStore';
import { DEFAULT_LOCALE } from 'utils/locales';
import RegisterPage from '../index';
import 'utils/__tests__/__mocks__/matchMedia';

describe('<RegisterPage />', () => {
  const history = createMemoryHistory();
  const store = configureStore({}, history);

  it('should render its registerPage', () => {
    const {
      container: { firstChild },
    } = render(
      <Provider store={store}>
        <IntlProvider locale={DEFAULT_LOCALE}>
          <ConnectedRouter history={history}>
            <HelmetProvider>
              <RegisterPage />
            </HelmetProvider>
          </ConnectedRouter>
        </IntlProvider>
      </Provider>,
    );

    expect(firstChild).toMatchSnapshot();
  });
});
