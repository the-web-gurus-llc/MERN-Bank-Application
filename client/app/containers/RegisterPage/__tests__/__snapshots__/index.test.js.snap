// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RegisterPage /> should render its registerPage 1`] = `
<header
  class="Headerstyle__StyledHeader-ae05xe-0 gxFPFE"
>
  <a
    href="/"
  >
    <img
      alt="Bank Application"
      class="Headerstyle__StyledImg-ae05xe-1 ggLcVg"
      height="100%"
      src="IMAGE_MOCK"
      width="180"
    />
  </a>
  <img
    alt="Adrian Pietrzak"
    class="Headerstyle__StyledImg-ae05xe-1 ggLcVg"
    height="100%"
    src="IMAGE_MOCK"
    width="50"
  />
</header>
`;
